erpnext/erpnext_integrations/doctype/taxjar_settings/taxjar_settings.py

```diff
@@ -4,9 +4,98 @@
 
 from __future__ import unicode_literals
 
-# import frappe
+import json
+import os
+
+import frappe
+from frappe.custom.doctype.custom_field.custom_field import create_custom_fields
 from frappe.model.document import Document
+from frappe.permissions import add_permission, update_permission_property
+
+from erpnext.erpnext_integrations.taxjar_integration import get_client
 
 
 class TaxJarSettings(Document):
-	pass
+
+	def on_update(self):
+		TAXJAR_CREATE_TRANSACTIONS = frappe.db.get_single_value("TaxJar Settings", "taxjar_create_transactions")
+		TAXJAR_CALCULATE_TAX = frappe.db.get_single_value("TaxJar Settings", "taxjar_calculate_tax")
+		TAXJAR_SANDBOX_MODE = frappe.db.get_single_value("TaxJar Settings", "is_sandbox")
+		fields_hidden = 0
+
+		custom_fields = []
+		for dt in ['Item', 'Sales Invoice Item']:
+			doc = frappe.get_doc('Custom Field', {'dt': dt, 'fieldname':'product_tax_category'})
+			custom_fields.append(doc)
+			fields_hidden = doc.get('hidden') if doc else 0
+		fields_already_exist = True if custom_fields else False
+
+
+		if (TAXJAR_CREATE_TRANSACTIONS or TAXJAR_CALCULATE_TAX or TAXJAR_SANDBOX_MODE):
+			if not fields_already_exist:
+				add_product_tax_categories()
+				make_custom_fields()
+				add_permissions()
+				frappe.enqueue('erpnext.regional.united_states.setup.add_product_tax_categories', now=False)
+
+			elif fields_already_exist and fields_hidden:
+				toggle_tax_category_fields('1')
```

Comment:
This could be more readable as:

```suggestion
				toggle_tax_category_fields(toggle=True)
```

or

```suggestion
				toggle_tax_category_fields(hidden=True)
```

after changing the argument name in the `toggle_tax_category_fields` function